main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
BTW, CTags did something odd (for bad reasons if you ask me) with `\r`s without following `\n`: it handed them as bare characters, not terminating the line.  However, various tools do consider them line ends on GNU/Linux at least.